File: src/frontend/src/app/model/project-fields/CheckBoxContainerDto.ts
import { CheckBoxDto } from './CheckBoxDto';
import { ProjectFieldDto } from './ProjectFieldDto';

export class CheckBoxContainerDto extends ProjectFieldDto {
    elements: Array<CheckBoxDto> = [];

    constructor(id: number, fieldType: string, fieldName: string, isRequired: boolean, elements: Array<CheckBoxDto>) {
        super(id, fieldType, fieldName, isRequired);
        this.elements = elements;
    }
}

File: src/main/java/scrumweb/dto/projectfield/CheckBoxContainerDto.java
package scrumweb.dto.projectfield;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@NoArgsConstructor
@Getter @Setter
public class CheckBoxContainerDto extends ProjectFieldDto {
    private Set<CheckBoxDto> elements;

    public CheckBoxContainerDto(Long id, String fieldType, String fieldName, Boolean isRequired, Set<CheckBoxDto> elements) {
        super(id, fieldType, fieldName, isRequired);
        this.elements = elements;
    }
}
